From 954158f9265ac07fdc3bf32148c92bdb65f72e06 Mon Sep 17 00:00:00 2001
From: Jason King <jason.king@joyent.com>
Date: Tue, 8 May 2018 21:44:21 -0500
Subject: [PATCH] OS-6924 Double-free in crypto tests on failure Reviewed by:
 Dan McDonald <danmcd@joyent.com> Reviewed by: Mike Zeller
 <mike.zeller@joyent.com> Approved by: Mike Zeller <mike.zeller@joyent.com>

---
 usr/src/test/crypto-tests/tests/common/cryptotest_kcf.c  | 3 +--
 usr/src/test/crypto-tests/tests/common/cryptotest_pkcs.c | 2 --
 2 files changed, 1 insertion(+), 4 deletions(-)

diff --git a/usr/src/test/crypto-tests/tests/common/cryptotest_kcf.c b/usr/src/test/crypto-tests/tests/common/cryptotest_kcf.c
index 260821dcac..f19aa0b7c2 100644
--- a/usr/src/test/crypto-tests/tests/common/cryptotest_kcf.c
+++ b/usr/src/test/crypto-tests/tests/common/cryptotest_kcf.c
@@ -11,6 +11,7 @@
 
 /*
  * Copyright 2015 Nexenta Systems, Inc.  All rights reserved.
+ * Copyright 2018, Joyent, Inc.
  */
 
 #include <fcntl.h>
@@ -134,7 +135,6 @@ get_mech_info(crypto_op_t *op)
 	    (uint_t *)&get_number, "get_mech_info") != CRYPTO_SUCCESS) {
 		(void) fprintf(stderr, "failed to resolve mechanism name %s\n",
 		    op->mechname);
-		(void) cryptotest_close(op);
 		return (CTEST_NAME_RESOLVE_FAILED);
 	}
 	op->mech = get_number.pn_internal_number;
@@ -158,7 +158,6 @@ get_hsession_by_mech(crypto_op_t *op)
 		(void) fprintf(stderr,
 		    "could not find provider for mechanism %llu\n",
 		    mech.mech_type);
-		(void) cryptotest_close(op);
 		return (CTEST_MECH_NO_PROVIDER);
 	}
 
diff --git a/usr/src/test/crypto-tests/tests/common/cryptotest_pkcs.c b/usr/src/test/crypto-tests/tests/common/cryptotest_pkcs.c
index db15968b49..0506110e3c 100644
--- a/usr/src/test/crypto-tests/tests/common/cryptotest_pkcs.c
+++ b/usr/src/test/crypto-tests/tests/common/cryptotest_pkcs.c
@@ -104,7 +104,6 @@ get_mech_info(crypto_op_t *op)
 		cryptotest_error("get_mech_info", rv);
 		(void) fprintf(stderr, "failed to resolve mechanism name %s\n",
 		    op->mechname);
-		(void) cryptotest_close(op);
 		return (CTEST_NAME_RESOLVE_FAILED);
 	}
 	return (rv);
@@ -121,7 +120,6 @@ get_hsession_by_mech(crypto_op_t *op)
 		(void) fprintf(stderr,
 		    "could not find provider for mechanism %lu\n",
 		    op->mech);
-		(void) cryptotest_close(op);
 		return (CTEST_MECH_NO_PROVIDER);
 	}
 	return (rv);
-- 
2.21.0

